pkg/proxy/iptables/proxier.go

```diff
@@ -290,6 +290,7 @@ func NewProxier(ipt utiliptables.Interface,
 	serviceHealthServer := healthcheck.NewServiceHealthServer(hostname, recorder)
 
 	isIPv6 := ipt.IsIPv6()
+	nodePortAddresses, _ = utilproxy.FilterIncorrectCIDRVersion(nodePortAddresses, isIPv6)
```

Comment:
are there tests that exercise this scenario?

Reply:
Regression tests with ipv4 address, yes. Ipv6 addresses, no (as usual), mixed addresses, no.
I assume the library functions "FilterIncorrectCIDRVersion" has it's own tests.

Reply:
The FilterIncorrectCIDRVersion does have tests https://github.com/kubernetes/kubernetes/blob/4d0e86f0b8d1eae00a202009858c8739e4c9402e/pkg/proxy/util/utils_test.go#L541

It's being used to discriminate IP address of the load balancer services

https://github.com/kubernetes/kubernetes/blob/0a6c826d3e92dae8f20d6199d0ac7deeca9eed71/pkg/proxy/service.go#L157-L165

For the NewProxier testing there is  a TODO

https://github.com/kubernetes/kubernetes/blob/0a6c826d3e92dae8f20d6199d0ac7deeca9eed71/pkg/proxy/iptables/proxier_test.go#L350-L351

Reply:
filtering here and silently discarding invalid NodePortAddresses doesn't seem great... I would have expected to do the filtering in the NewDualStackProxier methods:

```
func NewDualStackProxier(
...
ipv4NodePortAddresses, ipv6NodePortAddresses := utilproxy.FilterIncorrectCIDRVersion(nodePortAddresses, false)
ipv4Proxier, err := NewProxier(..., ipv4NodePortAddresses, ...)
ipv6Proxier, err := NewProxier(..., ipv6NodePortAddresses, ...)
```

In NewProxier, if nodePortAddresses of the wrong family are passed in, that seems like something we should error on (that would catch the case where a single-stack proxier was created and --nodeport-addresses was misconfigured with the wrong family)

Reply:
I did that first and moved to `NewProxier` :smiley: I can move the check back. It it enough to log a fault _and_ filter the addresses to avoid the faulty entries reported in https://github.com/kubernetes/kubernetes/issues/89923? I does not seem right to crash `kube-proxy` in case of a misconfigured NodePortAddresses.

Reply:
> I did that first and moved to `NewProxier`  I can move the check back.

it was my fault to be honest https://github.com/kubernetes/kubernetes/pull/89998#issuecomment-611430756 :smile: